.github/workflows/ci.yml

```diff
@@ -214,3 +218,79 @@ jobs:
         make -j2
         ctest -j2 --output-on-failure
       working-directory: examples
+
+    - name: Build Documentation
+      if: matrix.build-documentation
+      run: make -j2 docs-gh
+      working-directory: build
+    
+    - name: Store Documentation
+      if: matrix.build-documentation
+      uses: actions/upload-artifact@v2
+      with:
+        name: documentation
+        path: build/docs/sphinx-gh/
+
+  upload-docs:
+    if: github.event_name == 'push' || github.event_name == 'pull_request_target'
+    name: upload-docs
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    needs: build
```

Comment:
Can we make this dependent on `ubuntu:production` only? Then we don't have to wait with the upload until all builds are done.

Reply:
It seems this is not possible: https://github.com/CVC4/CVC4/actions/runs/815832121
Also, I've found nothing in the documentation about requiring a single job from a matrix